mirgecom/array_context.py

```diff
@@ -176,6 +180,10 @@ def _check_gpu_oversubscription(actx: ArrayContext) -> None:
     Only works with CUDA devices currently due to the use of the
     PCI_DOMAIN_ID_NV extension.
     """
+    import sys
+    if "mpi4py" not in sys.modules:
+        return
+
```

Comment:
Would it make sense to add an `actx_class_is_distributed` or `actx_class_is_mpi` function and only call `_check_cache_dirs_node`/`_check_gpu_oversubscription` if it returns `True`?

Reply:
Thanks! What do you think of [16a3a6a](https://github.com/illinois-ceesd/mirgecom/pull/1060/commits/16a3a6a9560db47c01b18bcac13ad366a10f8c04)?